crypto/cipher/aes256cbc/decrypter.go

```diff
@@ -49,42 +48,41 @@ func (d Decrypter) Decrypt(data mc.EncryptedContent) (mc.PlainContent, error) {
 func decryptEncryptedData(privKey crypto.PrivateKey, data *encryptedData) ([]byte, error) {
 	tmpEphemeralPublicKey, err := secp256k1.PublicKeyFromBytes(data.EphemeralPublicKey)
 	if err != nil {
-		return nil, errors.WithMessage(err, "could not convert ephemeralPublicKey")
+		return nil, mc.ErrDecrypt
 	}
 
 	ephemeralPublicKey, err := tmpEphemeralPublicKey.(*secp256k1.PublicKey).ECIES()
 	if err != nil {
-		return nil, errors.WithMessage(err, "could not convert to ecies")
+		return nil, mc.ErrDecrypt
 	}
 
 	recipientPrivKey, err := asPrivateECIES(privKey)
 	if err != nil {
-		return nil, err
+		return nil, mc.ErrDecrypt
 	}
 
 	sharedSecret, err := deriveSharedSecret(ephemeralPublicKey, recipientPrivKey)
 	if err != nil {
-		return nil, errors.WithMessage(err, "could not derive shared secret")
+		return nil, mc.ErrDecrypt
 	}
 
 	macKey, encryptionKey := generateMacKeyAndEncryptionKey(sharedSecret)
 	mac, err := generateMac(macKey, data.InitializationVector, *ephemeralPublicKey, data.Ciphertext)
 
 	if err != nil {
-		return nil, errors.WithMessage(err, "generateMac failed")
```

Comment:
I don't think it's a good idea to mask all errors with the same error. It's okay to return the same error type, but it should not be the same error message for different types of errors, the actual error is getting masked here. It'll be harder to debug the reason just by seeing "error in decryption".

Reply:
I believe that in the future we will create a variable for each type of error, making the code more maintainable.

Reply:
@hasitpbhatt thank you for taking the time to leave a review on this PR. Sorry for the delay getting back to you I was off for a while. I would love to hear about how you found out about Mailchain and if there is anything I can do to help you. Please feel free to comment here or join our [slack channel](https://github.com/mailchain/mailchain#get-involved)

You raise an excellent point about the outcome from a developer perspective. The reason for masking the error is to reduce the information an attacker might get from an error.  This does need more thought overall so I have opened an issues #532 to discuss this. 

In the meantime @developerfred I will merge this change as it resolves the issue as originally defined. Once #532 is resolved we will come back to this